cli/command/service/opts.go

```diff
@@ -506,35 +586,122 @@ func addServiceFlags(flags *pflag.FlagSet, opts *serviceOptions) {
-	flags.DurationVar(&opts.update.monitor, flagUpdateMonitor, time.Duration(0), "Duration after each task update to monitor for failure (ns|us|ms|s|m|h)")
+	updateDelayDesc := "Delay between updates (ns|us|ms|s|m|h)"
+	if !isUpdate {
+		defaultUpdateDelay := defaults.Service.Update.Delay
+		updateDelayDesc += fmt.Sprintf(" (default %s)", defaultUpdateDelay)
+	}
+	flags.DurationVar(&opts.update.delay, flagUpdateDelay, 0, updateDelayDesc)
+
+	updateMonitorDesc := "Duration after each task update to monitor for failure (ns|us|ms|s|m|h)"
+	if !isUpdate {
+		defaultUpdateMonitor, _ := gogotypes.DurationFromProto(defaults.Service.Update.Monitor)
+		updateMonitorDesc += fmt.Sprintf(" (default %s)", defaultUpdateMonitor)
+	}
+	flags.DurationVar(&opts.update.monitor, flagUpdateMonitor, 0, updateMonitorDesc)
 	flags.SetAnnotation(flagUpdateMonitor, "version", []string{"1.25"})
-	flags.StringVar(&opts.update.onFailure, flagUpdateFailureAction, "pause", `Action on update failure ("pause"|"continue"|"rollback")`)
-	flags.Var(&opts.update.maxFailureRatio, flagUpdateMaxFailureRatio, "Failure rate to tolerate during an update")
+
+	updateFailureActionDesc := `Action on update failure ("pause"|"continue"|"rollback")`
+	if !isUpdate {
+		defaultUpdateFailureAction := strings.ToLower(api.UpdateConfig_FailureAction_name[int32(defaults.Service.Update.FailureAction)])
+		updateFailureActionDesc += fmt.Sprintf(` (default "%s")`, defaultUpdateFailureAction)
+	}
+	flags.StringVar(&opts.update.onFailure, flagUpdateFailureAction, "", updateFailureActionDesc)
+
+	updateMaxFailureRatioDesc := "Failure rate to tolerate during an update"
+	if !isUpdate {
+		updateMaxFailureRatioDesc += fmt.Sprintf(" (default %f)", defaults.Service.Update.MaxFailureRatio)
+	}
+	flags.Var(&opts.update.maxFailureRatio, flagUpdateMaxFailureRatio, updateMaxFailureRatioDesc)
 	flags.SetAnnotation(flagUpdateMaxFailureRatio, "version", []string{"1.25"})
 
-	flags.Uint64Var(&opts.rollback.parallelism, flagRollbackParallelism, 1, "Maximum number of tasks rolled back simultaneously (0 to roll back all at once)")
+	var defaultRollbackParallelism uint64
+	if !isUpdate {
+		defaultRollbackParallelism = defaults.Service.Rollback.Parallelism
+	}
+	flags.Uint64Var(&opts.rollback.parallelism, flagRollbackParallelism, defaultRollbackParallelism, "Maximum number of tasks rolled back simultaneously (0 to roll back all at once)")
 	flags.SetAnnotation(flagRollbackParallelism, "version", []string{"1.28"})
-	flags.DurationVar(&opts.rollback.delay, flagRollbackDelay, time.Duration(0), "Delay between task rollbacks (ns|us|ms|s|m|h) (default 0s)")
+
+	rollbackDelayDesc := "Delay between task rollbacks (ns|us|ms|s|m|h)"
+	if !isUpdate {
+		defaultRollbackDelay := defaults.Service.Rollback.Delay
+		rollbackDelayDesc += fmt.Sprintf(" (default %s)", defaultRollbackDelay)
+	}
+	flags.DurationVar(&opts.rollback.delay, flagRollbackDelay, 0, rollbackDelayDesc)
```

Comment:
`DurationVar` is a little weird. IIRC it shows the default value in the CLI help, but only if it's nonzero. I'd like us to always show the default on `service create`, even if it's zero, and never show it on `service update`. So it made sense to manipulate the flag description instead of providing a default value. If we did both, the default value would appear twice when the default is nonzero.

I can add a comment explaing this.